### ISSUE TITLE
[feature] Prune old remote posts completely

### ISSUE THREAD

**Issue by gw1urf** [superseriousbusiness/gotosocial#1919]
### Is your feature request related to a problem ?

Disk usage of database grows without bound, filling up with statuses that are unlikely ever to be revisited.

### Describe the solution you'd like.

It would be nice to be able to set a time after which remotely originated posts will be completely deleted from the database (along with any media) if they've had no local interactions (likes, boosts, comments).

### Describe alternatives you've considered.

NONE

### Additional context.

I've been running a single user instance for ~8 months. The media pruning works great, but my SQLite database is now > 2GBytes and becoming one of the biggest things on my VPS. The "statuses" table has nearly 700,000 entries.

It seems to me that if a remote status has never had a comment, boost, bookmark or like by any local account, it could be pruned from the database after a sensible time. I'm unlikely ever to scroll back even 2 weeks through my timeline, so if the items I never interacted with disappeared, I wouldn't notice.

**Comment by tsmethurst** [comment]
related: https://github.com/superseriousbusiness/gotosocial/issues/1442

**Comment by xmgz** [comment]
yes some kind of purge/autoclean setting for both local and remote content would be nice to have as an **opt in** feature so anybody could decide what's best for their instance.

Posts without external interactions older than $TIME would be a nice first implementation. Both DB and Multimedia folder size (important for small VPS) would benefit.


**thank you**



(reached here because I **need** to prune old Multimedia safely, but I don't want to mess with DB problems)

**Comment by NyaaaWhatsUpDoc** [comment]
@xmgz pruning of old statuses (and possibly, accounts) is planned for some point in the future. though media can already pruned, there's an admin action for it you can access through the front-end settings UI, a CLI command, and an operation that runs for it every night with a configurable max. number of days worth of remote media to cache.

**Comment by daenney** [comment]
We've migrated to Codeberg. Your issue has been imported as https://codeberg.org/superseriousbusiness/gotosocial/issues/1919.
